PriceKataCalc/Program.cs

```diff
@@ -89,19 +104,55 @@ private double UPCDiscount()
             else return 0;
         }
 
-        internal object calcAllDiscount()
+        public double calcAllDiscount()
         {
             return this.calcDiscount() + this.calcUPCDiscount();   
         }
+
+        public double calcCosts()
+        {
+            double sum = 0;
+            foreach(cost i in this.Costs)
+            {
+                if (i.isPercentage) sum += this.price;
```

Comment:
Why should the product know if the cost percentage or not? This is an example of a leaky abstraction